build-system/tasks/runtime-test.js

```diff
@@ -18,27 +18,38 @@ var argv = require('minimist')(process.argv.slice(2));
 var gulp = require('gulp-help')(require('gulp'));
 var Karma = require('karma').Server;
 var config = require('../config');
-var karmaConfig = config.karma;
-var extend = require('util')._extend;
 var fs = require('fs');
 var path = require('path');
 var util = require('gulp-util');
 var webserver = require('gulp-webserver');
 var app = require('../test-server').app;
 
+var karmaDefault = {
+  configFile: path.resolve('build-system/tasks/karma.conf.js'),
+  // "client" has to be specified here but not in configFile,
+  // because we want to change its value based on flags.
+  // The whole "client" object will get overridden if specified in config file.
+  client: {
```

Comment:
the eventual config passed to Karma consists of 1) a config file path, 2) some overrides.
If you specify `client` in the config file, and in L152 we introduce an override, which  simply overrides the whole `client` object.